### PR TITLE
Implement in memory logic

### PR DESCRIPTION
**Pull request by rapasoft**
@misamoto Pls review and merge

### CODE REVIEW

File: src/main/java/no/itera/bloggingplatform/repository/memory/DefaultInMemoryRepository.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import no.itera.bloggingplatform.repository.BasicRepository;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public abstract class DefaultInMemoryRepository<T extends Peristable> implements BasicRepository<T> {
+
+    final Map<Long, T> inMemoryCache = new HashMap<>();
```
Comment (misamoto) [inline src/main/java/no/itera/bloggingplatform/repository/memory/DefaultInMemoryRepository.java]: Can we have this map private? 
Values are available via `DefaultInMemoryRepository#readAll()`

File: src/main/java/no/itera/bloggingplatform/repository/BasicRepository.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    List<T> readAll();
+
+    /**
+     * Updates object in storage with new value based on key
+     *
+     * @param id       key of object to be replaced
+     * @param toUpdate replacement to be used
+     * @return updated object with correct key assigned
+     */
+    T update(Long id, T toUpdate);
```
Comment (misamoto) [inline src/main/java/no/itera/bloggingplatform/repository/BasicRepository.java]: Don't we expect the entity to contain the right `id`?

File: src/main/java/no/itera/bloggingplatform/model/Peristable.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package no.itera.bloggingplatform.model;
+
+public interface Peristable {
```
Comment (misamoto) [inline src/main/java/no/itera/bloggingplatform/model/Peristable.java]: Typo in `Peristable` .. missing `s` :)

File: src/main/java/no/itera/bloggingplatform/model/Peristable.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+    Long getKey();
+
+    void setKey(Long id);
+
```
Comment (misamoto) [inline src/main/java/no/itera/bloggingplatform/model/Peristable.java]: How about adding `boolean exists()`?
We'd be able to remove checks like these from service layer:
```
if (category.getKey() != null) {
            return category;
}
```
That'd make it a bit cleaner.
I'd expect the repo/persistence layer to deal with it so I propose we move the check to `DefaultInMemoryRepository#create` : 
```
    @Override
    public T create(T toPersist) {
        if (toPersist.exists()) {
            return toPersist;
        }
        toPersist.setKey(SimpleIdGenerator.generate());
        inMemoryCache.put(toPersist.getKey(), toPersist);
        return toPersist;
    }
```

File: src/main/java/no/itera/bloggingplatform/repository/memory/DefaultInMemoryRepository.java (hunk 1 above)
Comment (rapasoft) [reply src/main/java/no/itera/bloggingplatform/repository/memory/DefaultInMemoryRepository.java]: Sure!

File: src/main/java/no/itera/bloggingplatform/repository/BasicRepository.java (hunk 2 above)
Comment (rapasoft) [reply src/main/java/no/itera/bloggingplatform/repository/BasicRepository.java]: Yes, that was the intention. E.g. if the instance of the object with `id` does not exist it should fail to update. (this is not implemented yet, but will do)

File: src/main/java/no/itera/bloggingplatform/model/Peristable.java (hunk 3 above)
Comment (rapasoft) [reply src/main/java/no/itera/bloggingplatform/model/Peristable.java]: D'oh!

File: src/main/java/no/itera/bloggingplatform/model/Peristable.java (hunk 4 above)
Comment (rapasoft) [reply src/main/java/no/itera/bloggingplatform/model/Peristable.java]: I agree with the `exists()` method, but I would slightly adjust contract. If objects exist, then it should not try to return the same object, but rather throw an exception

File: src/main/java/no/itera/bloggingplatform/repository/BasicRepository.java (hunk 2 above)
Comment (misamoto) [reply src/main/java/no/itera/bloggingplatform/repository/BasicRepository.java]: Sure, either way, the `Long id` param is not really necessary:)
I mean, where would we get that `id`, but from fetched (persisted) entity itself?

File: src/main/java/no/itera/bloggingplatform/model/Peristable.java (hunk 4 above)
Comment (misamoto) [reply src/main/java/no/itera/bloggingplatform/model/Peristable.java]: Yup, even better

**Pull request closed without merging**